From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Thu, 2 Feb 2023 09:59:24 +0100
Subject: fix: alter coreModuleRegExp to prevent arm crash

For some reason the choice to use /^ {4} instead of four spaces is causing a crash
in Windows on Arm in V8 Regex parsing. Fix this by replacing with four actual
spaces to maintain functionality without a crash.

This will be reported upstream as it reproduces there.

diff --git a/lib/internal/util/inspect.js b/lib/internal/util/inspect.js
index 6174d62bb11643949bfb041db36b94f1998d0df9..97c9bc0c5af55b8961cd99ce24844664393b0dd5 100644
--- a/lib/internal/util/inspect.js
+++ b/lib/internal/util/inspect.js
@@ -206,7 +206,7 @@ const strEscapeSequencesReplacerSingle = /[\x00-\x1f\x5c\x7f-\x9f]|[\ud800-\udbf
 const keyStrRegExp = /^[a-zA-Z_][a-zA-Z_0-9]*$/;
 const numberRegExp = /^(0|[1-9][0-9]*)$/;
 
-const coreModuleRegExp = /^ {4}at (?:[^/\\(]+ \(|)node:(.+):\d+:\d+\)?$/;
+const coreModuleRegExp = /^    at (?:[^/\\(]+ \(|)node:(.+):\d+:\d+\)?$/;
 const nodeModulesRegExp = /[/\\]node_modules[/\\](.+?)(?=[/\\])/g;
 
 const classRegExp = /^(\s+[^(]*?)\s*{/;
